### ISSUE TITLE
DISABLED test_backward_ddp_outside (__main__.ProcessGroupDdpUnderDistAutogradTestWithSpawn)

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#40378]
## 🐛  test_backward_ddp_outside is flaky

See for ex. https://app.circleci.com/pipelines/github/pytorch/pytorch/183902/workflows/9bbc51db-4c8e-4d47-92ea-a43fe60ac4ce/jobs/5952141/steps

```
Jun 22 00:59:56 ======================================================================
Jun 22 00:59:56 ERROR [61.552s]: test_backward_ddp_outside (__main__.TestDdpUnderDistAutogradWrapper)
Jun 22 00:59:56 ----------------------------------------------------------------------
Jun 22 00:59:56 Traceback (most recent call last):
Jun 22 00:59:56   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 204, in wrapper
Jun 22 00:59:56     self._join_processes(fn)
Jun 22 00:59:56   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 306, in _join_processes
Jun 22 00:59:56     self._check_return_codes(elapsed_time)
Jun 22 00:59:56   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 339, in _check_return_codes
Jun 22 00:59:56     raise RuntimeError(error)
Jun 22 00:59:56 RuntimeError: Processes 5 exited with error code 10
Jun 22 00:59:56 
Jun 22 00:59:56 ----------------------------------------------------------------------
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528 @osalpekar @jjlilley

**Comment by pritamdamania87** [comment]
Fixed by https://github.com/pytorch/pytorch/pull/40495

**Comment by mrshenli** [comment]
Reopen this issue as it occurs again: 
https://app.circleci.com/pipelines/github/pytorch/pytorch/195326/workflows/b4d67acd-38e4-4558-936b-b89b5840957d/jobs/6450811/steps

```
ul 31 02:20:26   test_backward_ddp_outside (__main__.DdpUnderDistAutogradTest) ... 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:313 INFO p:process 0 t:MainThread: Running the trainer #0...
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:313 INFO p:process 1 t:MainThread: Running the trainer #1...
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:313 INFO p:process 2 t:MainThread: Running the trainer #2...
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:315 INFO p:process 0 t:MainThread: Initing trainer process group by trainer #0 with ranks [0, 1, 2, 3]
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:315 INFO p:process 2 t:MainThread: Initing trainer process group by trainer #2 with ranks [0, 1, 2, 3]
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:315 INFO p:process 1 t:MainThread: Initing trainer process group by trainer #1 with ranks [0, 1, 2, 3]
Jul 31 02:20:26 2020-07-31 02:20:26,509 ddp_under_dist_autograd_test.py:300 INFO p:process 4 t:MainThread: The remote worker is running.
Jul 31 02:20:26 2020-07-31 02:20:26,511 ddp_under_dist_autograd_test.py:332 INFO p:process 5 t:MainThread: Running the master process...
Jul 31 02:20:26 2020-07-31 02:20:26,511 ddp_under_dist_autograd_test.py:313 INFO p:process 3 t:MainThread: Running the trainer #3...
Jul 31 02:20:26 2020-07-31 02:20:26,512 ddp_under_dist_autograd_test.py:315 INFO p:process 3 t:MainThread: Initing trainer process group by trainer #3 with ranks [0, 1, 2, 3]
Jul 31 02:20:26 2020-07-31 02:20:26,514 ddp_under_dist_autograd_test.py:323 INFO p:process 3 t:MainThread: Waiting for shutdown signal on trainer #3...
Jul 31 02:20:26 2020-07-31 02:20:26,515 ddp_under_dist_autograd_test.py:346 INFO p:process 5 t:MainThread: Created remote rrefs on master
Jul 31 02:20:26 2020-07-31 02:20:26,533 ddp_under_dist_autograd_test.py:95 INFO p:process 4 t:Dummy-1: Initing RemoteEM with 2 3
Jul 31 02:20:26 2020-07-31 02:20:26,534 ddp_under_dist_autograd_test.py:121 INFO p:process 4 t:Dummy-2: Initing RemoteNet with 5 3
Jul 31 02:20:26 2020-07-31 02:20:26,535 ddp_under_dist_autograd_test.py:323 INFO p:process 0 t:MainThread: Waiting for shutdown signal on trainer #0...
Jul 31 02:20:26 2020-07-31 02:20:26,540 ddp_under_dist_autograd_test.py:323 INFO p:process 2 t:MainThread: Waiting for shutdown signal on trainer #2...
Jul 31 02:20:26 2020-07-31 02:20:26,541 ddp_under_dist_autograd_test.py:323 INFO p:process 1 t:MainThread: Waiting for shutdown signal on trainer #1...
Jul 31 02:21:26 ERROR:root:Caught exception: 
Jul 31 02:21:26 Traceback (most recent call last):
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 213, in wrapper
Jul 31 02:21:26     fn()
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/dist_utils.py", line 94, in new_test_method
Jul 31 02:21:26     return_value = old_test_method(self, *arg, **kwargs)
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 426, in test_backward_ddp_outside
Jul 31 02:21:26     self._do_test(DdpMode.OUTSIDE)
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 410, in _do_test
Jul 31 02:21:26     self._master_process(ddp_mode)
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 347, in _master_process
Jul 31 02:21:26     self.do_test_on_master(ddp_mode, remote_em_rref, remote_net_rref)
Jul 31 02:21:26   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 379, in do_test_on_master
Jul 31 02:21:26     ddp_grads, non_ddp_grads = future.wait()
Jul 31 02:21:26 RuntimeError: RPCErr:1:RPC ran for more than 60000 milliseconds and timed out.
Jul 31 02:21:26 exiting process with exit code: 10
Jul 31 02:21:26 Process 5 terminated with exit code 10, terminating remaining processes.
Jul 31 02:21:26 ERROR (61.712s)
```

**Comment by mrshenli** [comment]
closed by #45181
